### ISSUE TITLE
Refactor RuntimeOptions.cucumberFeatures out of RuntimeOptions

### ISSUE THREAD

**Issue by mpkorstanje** [cucumber/cucumber-jvm#1366]
Currently `RuntimeOptions` has too many responsibilities. In addition to managing the runtime options:
 * It parses command line options
 * It manages and instantiates plugins
 * It loads and instantiates cucumberFeatures

Ultimately `RuntimeOptions` should become an immutable data class that implements several different interfaces to be used by different consumers.

Solving this can be done in several steps. This is one such step:

`RuntimeOptions.cucumberFeatures` and related functions should be in their own `CucumberFeatures` class. The responsibility of emitting the `TestRunStarted` should be moved to the callers of `cucumberFeatures`.

So ultimately the dependency graph will be : 

```
RuntimeOptions <--- CucumberFeatures <--- Runtime (use constructor dependency injection)
EventBus       <---|
ResourceLoader <---|
```

**Comment by aslakhellesoy** [comment]
Sounds like a good plan. Do you think we could fit in [config](https://github.com/cucumber/cucumber/tree/master/config) to populate `RuntimeOptions` with values @mpkorstanje?

**Comment by mpkorstanje** [comment]
I'm making theses issues really small so any one can pick them up. Using config to populate run time options would be a separate task. There are a few more tasks that involve trimming down RuntimeOptions. After this is done implementing config would be easier.

**Comment by mpkorstanje** [comment]
Closed by 56c34773f9a5b3599b3f842fa49b13545db38204
